### ISSUE TITLE
Web of Things (WoT) WG (discussing)

### ISSUE THREAD

**Issue by mmccool** [w3c/lds-wg-charter#78]
(note: we are still reviewing and discussing charter, following will be revised after we gather group input on the specifics in the charter).

Web of Things (WoT) has also working on a canonical form (and signature mechanism) for WoT Thing Descriptions.  The canonical form is reasonably stable at this point; see https://w3c.github.io/wot-thing-description/#canonicalization-serialization-json.   

We were hoping to be able to base this on a standard JSON-LD canonicalization and signature mechanism, but the timing was not working out.  Hopefully we can get at least partially aligned with your direction and intercept it with WoT TD 2.0.  There are also some special concerns arising from TD peculiarities.  Regardless, the goal of having a way to canonicalize and sign JSON-LD is important.

**Comment by mmccool** [comment]
We discussed this in our main call on May 12 and agreed that
* We wish to strongly support the formation and goals of this group
* We probably need to proceed in the short form with our own canonicalization and signature approach
* We need input and advice from the LDS to align as close as possible with the (expected) form of the LDS based on current drafts (eg LD-PROOFS)
* We need a plan to align revisions of the TD spec with the final form of the LDS specification and generally need to closely collaborate

**Comment by iherman** [comment]
@mmccool are there any extra statements you want to add to the liaison text that we have put into the charter for the WoT WG?

**Comment by mmccool** [comment]
The liaison text is fine, and thank you for including it.  I think we probably have more feedback on other deliverables, in particular, we definitely want to see signatures supported in JSON-LD, and be computable without having to pull the JSON-LD into an RDF database (e.g. computable using only a JSON processor).   I know the focus here is on RDF but this requirement arises from the need to validate signatures on smaller devices.

The links above have a number of other requirements but I don't think that one was explicitly stated; too "obvious" ;)

**Comment by mmccool** [comment]
I also found the LD-PROOFs draft to be fairly open-ended with support for many kinds of proofs and encryption suites.  I expect that for WoT we will probably want to extract a narrower, finite set to make sure that finite testable implementations are possible.

**Comment by iherman** [comment]
> I think we probably have more feedback on other deliverables, in particular, we definitely want to see signatures supported in JSON-LD, and be computable without having to pull the JSON-LD into an RDF database

TL;DR: this should not be a problem.

Longer: the signatures will be defined on RDF Datasets in general, independently of a particular serialization. Expressing them in JSON-LD, which is, after all, a serialization of RDF Datasets, is therefore clearly in scope. Furthermore, the charter explicitly refers to a proposed JSON-LD `@context` that can be used by applications.

**Comment by mmccool** [comment]
Not sure the "be computable without having to pull the JSON-LD into an RDF database" part is correct, however, since it seems the proposed LDS canonical form is a sorted set of RDF n-tuples (which obviously requires RDF processing).  In the meantime we are proceeding with a definition of an XML Signature/JWS-based approach, with a canonical JSON form using our own set of rules and JCS; see https://github.com/w3c/wot-thing-description/pull/1151.  In the long run, I can see systems using either these signatures (based on JSON) or the LDS (based on RDF).

**Comment by pchampin** [comment]
Dear @mmccool,
the proposed charter has been substantially changed since you submitted your expression of support. More specifically, its scoped has been narrowed down, to address some concerns raised in the Semantic Web community. See also [Ivan Herman's explanation of those changes on the SebWeb mailing list](https://lists.w3.org/Archives/Public/semantic-web/2021Oct/0020.html).

We kindly ask you to review [the new version](https://w3c.github.io/rch-wg-charter/), and either
* renew your expression of interest (by a simple reply to this thread), or
* to close this issue if you do not wish to renew it.

Thanks in advance

**Comment by pchampin** [comment]
up?
